### ISSUE TITLE
Sending SMS does not work

### ISSUE THREAD

**Issue by PaleoCrafter** [gammu/gammu#216]
I am trying to setup my Raspberry Pi together with a Huawei E169 stick to send SMS for a monitoring task. Gammu seems to recognize the device just fine and I can enter the PIN just fine, so the communication with the device is working. I tried sending an SMS but I got the following message:
`Sending SMS 1/1....waiting for network answer..error 500, message reference=-1`

I've seen this issue elsewhere, but either the solutions did not work for me or there were no answers to the issues. Hence, I file this report, hoping to get answers.
I've also tried running the `networkinfo` command, but that only yields `requesting network`, so it seems to be some issue with the network connection.

Further details about my device are here:
```
Device               : /dev/ttyUSB0
Manufacturer         : Huawei
Model                : E169 (E169)
Firmware             : 11.314.13.51.156
IMEI                 : XXXXXXXXXXXX
SIM IMSI             : XXXXXXXXXXXX
```

And here's the log from running `networkinfo`:
```
[Gammu            - 1.37.3]
[Connection       - "at"]
[Connection index - 0]
[Model type       - ""]
[Device           - "/dev/ttyUSB0"]
[Running on       - Linux, kernel 4.1.19-v7+ (#858 SMP Tue Mar 15 15:56:00 GMT 2016)]
Serial device: DTR is up, RTS is up, CAR is up, CTS is up
Setting speed to 115200
[Module           - "auto"]
Escaping SMS mode
SENDING frame type 0x00/length 0x02/2
1B |0D                                                          ..              
Sending simple AT command to wake up some devices
SENDING frame type 0x00/length 0x03/3
41A|54T|0D                                                      AT.             
1 "AT"
2 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x09/9
41A|54T|0D |0D |0A |4FO|4BK|0D |0A                              AT...OK..       
Enabling echo
SENDING frame type 0x00/length 0x05/5
41A|54T|45E|311|0D                                              ATE1.           
1 "ATE1"
2 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x0B/11
41A|54T|45E|311|0D |0D |0A |4FO|4BK|0D |0A                      ATE1...OK..     
Trying Motorola mode switch
SENDING frame type 0x00/length 0x0A/10
41A|54T|2B+|4DM|4FO|44D|45E|3D=|322|0D                          AT+MODE=2.      
1 "AT+MODE=2"
2 "COMMAND NOT SUPPORT"
Checking line: COMMAND NOT SUPPORT
AT reply state: 3
RECEIVED frame type 0x00/length 0x21/33
41A|54T|2B+|4DM|4FO|44D|45E|3D=|322|0D |0D |0A |43C|4FO|4DM|4DM AT+MODE=2...COMM
41A|4EN|44D|20 |4EN|4FO|54T|20 |53S|55U|50P|50P|4FO|52R|54T|0D  AND NOT SUPPORT.
0A                                                              .               
Seems not to be supported
Enabling CME errors
SENDING frame type 0x00/length 0x0A/10
41A|54T|2B+|43C|4DM|45E|45E|3D=|311|0D                          AT+CMEE=1.      
1 "AT+CMEE=1"
2 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x10/16
41A|54T|2B+|43C|4DM|45E|45E|3D=|311|0D |0D |0A |4FO|4BK|0D |0A  AT+CMEE=1...OK..
SENDING frame type 0x00/length 0x09/9
41A|54T|2B+|43C|53S|43C|53S|3F?|0D                              AT+CSCS?.       
1 "AT+CSCS?"
2 "+CSCS: "GSM""
3 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x1F/31
41A|54T|2B+|43C|53S|43C|53S|3F?|0D |0D |0A |2B+|43C|53S|43C|53S AT+CSCS?...+CSCS
3A:|20 |22"|47G|53S|4DM|22"|0D |0A |0D |0A |4FO|4BK|0D |0A      : "GSM"....OK.. 
SENDING frame type 0x00/length 0x0A/10
41A|54T|2B+|43C|53S|43C|53S|3D=|3F?|0D                          AT+CSCS=?.      
1 "AT+CSCS=?"
2 "+CSCS: ("IRA","GSM","UCS2")"
3 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x2F/47
41A|54T|2B+|43C|53S|43C|53S|3D=|3F?|0D |0D |0A |2B+|43C|53S|43C AT+CSCS=?...+CSC
53S|3A:|20 |28(|22"|49I|52R|41A|22"|2C,|22"|47G|53S|4DM|22"|2C, S: ("IRA","GSM",
22"|55U|43C|53S|322|22"|29)|0D |0A |0D |0A |4FO|4BK|0D |0A      "UCS2")....OK.. 
Chosen GSM as normal charset
Chosen UCS2 as unicode charset
Getting model
SENDING frame type 0x00/length 0x08/8
41A|54T|2B+|43C|47G|4DM|4DM|0D                                  AT+CGMM.        
1 "AT+CGMM"
2 "E169"
3 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x16/22
41A|54T|2B+|43C|47G|4DM|4DM|0D |0D |0A |45E|311|366|399|0D |0A  AT+CGMM...E169..
0D |0A |4FO|4BK|0D |0A                                          ..OK..          
[Model name: `E169']
[Model data: `E169']
[Model data: `E169']
[Connected model  - "E169"]
SENDING frame type 0x00/length 0x08/8
41A|54T|2B+|43C|47G|4DM|49I|0D                                  AT+CGMI.        
1 "AT+CGMI"
2 "huawei"
3 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x18/24
41A|54T|2B+|43C|47G|4DM|49I|0D |0D |0A |68h|75u|61a|77w|65e|69i AT+CGMI...huawei
0D |0A |0D |0A |4FO|4BK|0D |0A                                  ....OK..        
Manufacturer info received
[Manufacturer: Huawei]
Checking for OBEX support
SENDING frame type 0x00/length 0x0B/11
41A|54T|2B+|43C|50P|52R|4FO|54T|3D=|3F?|0D                      AT+CPROT=?.     
1 "AT+CPROT=?"
2 "COMMAND NOT SUPPORT"
Checking line: COMMAND NOT SUPPORT
AT reply state: 3
RECEIVED frame type 0x00/length 0x22/34
41A|54T|2B+|43C|50P|52R|4FO|54T|3D=|3F?|0D |0D |0A |43C|4FO|4DM AT+CPROT=?...COM
4DM|41A|4EN|44D|20 |4EN|4FO|54T|20 |53S|55U|50P|50P|4FO|52R|54T MAND NOT SUPPORT
0D |0A                                                          ..              
Checking for SYNCML/OBEX support
SENDING frame type 0x00/length 0x0C/12
41A|54T|2B+|53S|59Y|4EN|43C|4DM|4CL|3D=|3F?|0D                  AT+SYNCML=?.    
1 "AT+SYNCML=?"
2 "COMMAND NOT SUPPORT"
Checking line: COMMAND NOT SUPPORT
AT reply state: 3
RECEIVED frame type 0x00/length 0x23/35
41A|54T|2B+|53S|59Y|4EN|43C|4DM|4CL|3D=|3F?|0D |0D |0A |43C|4FO AT+SYNCML=?...CO
4DM|4DM|41A|4EN|44D|20 |4EN|4FO|54T|20 |53S|55U|50P|50P|4FO|52R MMAND NOT SUPPOR
54T|0D |0A                                                      T..             
SENDING frame type 0x00/length 0x0D/13
41A|54T|24$|54T|53S|53S|50P|43C|53S|57W|3D=|3F?|0D              AT$TSSPCSW=?.   
1 "AT$TSSPCSW=?"
2 "COMMAND NOT SUPPORT"
Checking line: COMMAND NOT SUPPORT
AT reply state: 3
RECEIVED frame type 0x00/length 0x24/36
41A|54T|24$|54T|53S|53S|50P|43C|53S|57W|3D=|3F?|0D |0D |0A |43C AT$TSSPCSW=?...C
4FO|4DM|4DM|41A|4EN|44D|20 |4EN|4FO|54T|20 |53S|55U|50P|50P|4FO OMMAND NOT SUPPO
52R|54T|0D |0A                                                  RT..            
[Module           - "A2D|iPAQ|at|M20|S25|MC35|TC35|C35i|S65|S300|5110|5130|5190|5210|6110|6130|6150|6190|6210|6250|6310|6310i|6510|7110|8210|8250|8290|8310|8390|8850|8855|8890|8910|9110|9210"]
Escaping SMS mode
SENDING frame type 0x00/length 0x02/2
1B |0D                                                          ..              
Sending simple AT command to wake up some devices
SENDING frame type 0x00/length 0x03/3
41A|54T|0D                                                      AT.             
1 "AT"
2 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x09/9
41A|54T|0D |0D |0A |4FO|4BK|0D |0A                              AT...OK..       
Enabling echo
SENDING frame type 0x00/length 0x05/5
41A|54T|45E|311|0D                                              ATE1.           
1 "ATE1"
2 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x0B/11
41A|54T|45E|311|0D |0D |0A |4FO|4BK|0D |0A                      ATE1...OK..     
Trying Motorola mode switch
SENDING frame type 0x00/length 0x0A/10
41A|54T|2B+|4DM|4FO|44D|45E|3D=|322|0D                          AT+MODE=2.      
1 "AT+MODE=2"
2 "COMMAND NOT SUPPORT"
Checking line: COMMAND NOT SUPPORT
AT reply state: 3
RECEIVED frame type 0x00/length 0x21/33
41A|54T|2B+|4DM|4FO|44D|45E|3D=|322|0D |0D |0A |43C|4FO|4DM|4DM AT+MODE=2...COMM
41A|4EN|44D|20 |4EN|4FO|54T|20 |53S|55U|50P|50P|4FO|52R|54T|0D  AND NOT SUPPORT.
0A                                                              .               
Seems not to be supported
Enabling CME errors
SENDING frame type 0x00/length 0x0A/10
41A|54T|2B+|43C|4DM|45E|45E|3D=|311|0D                          AT+CMEE=1.      
1 "AT+CMEE=1"
2 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x10/16
41A|54T|2B+|43C|4DM|45E|45E|3D=|311|0D |0D |0A |4FO|4BK|0D |0A  AT+CMEE=1...OK..
SENDING frame type 0x00/length 0x09/9
41A|54T|2B+|43C|53S|43C|53S|3F?|0D                              AT+CSCS?.       
1 "AT+CSCS?"
2 "+CSCS: "GSM""
3 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x1F/31
41A|54T|2B+|43C|53S|43C|53S|3F?|0D |0D |0A |2B+|43C|53S|43C|53S AT+CSCS?...+CSCS
3A:|20 |22"|47G|53S|4DM|22"|0D |0A |0D |0A |4FO|4BK|0D |0A      : "GSM"....OK.. 
SENDING frame type 0x00/length 0x0A/10
41A|54T|2B+|43C|53S|43C|53S|3D=|3F?|0D                          AT+CSCS=?.      
1 "AT+CSCS=?"
2 "+CSCS: ("IRA","GSM","UCS2")"
3 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x2F/47
41A|54T|2B+|43C|53S|43C|53S|3D=|3F?|0D |0D |0A |2B+|43C|53S|43C AT+CSCS=?...+CSC
53S|3A:|20 |28(|22"|49I|52R|41A|22"|2C,|22"|47G|53S|4DM|22"|2C, S: ("IRA","GSM",
22"|55U|43C|53S|322|22"|29)|0D |0A |0D |0A |4FO|4BK|0D |0A      "UCS2")....OK.. 
Chosen GSM as normal charset
Chosen UCS2 as unicode charset
SENDING frame type 0x00/length 0x08/8
41A|54T|2B+|43C|47G|4DM|49I|0D                                  AT+CGMI.        
1 "AT+CGMI"
2 "huawei"
3 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x18/24
41A|54T|2B+|43C|47G|4DM|49I|0D |0D |0A |68h|75u|61a|77w|65e|69i AT+CGMI...huawei
0D |0A |0D |0A |4FO|4BK|0D |0A                                  ....OK..        
Manufacturer info received
[Manufacturer: Huawei]
Checking for OBEX support
SENDING frame type 0x00/length 0x0B/11
41A|54T|2B+|43C|50P|52R|4FO|54T|3D=|3F?|0D                      AT+CPROT=?.     
1 "AT+CPROT=?"
2 "COMMAND NOT SUPPORT"
Checking line: COMMAND NOT SUPPORT
AT reply state: 3
RECEIVED frame type 0x00/length 0x22/34
41A|54T|2B+|43C|50P|52R|4FO|54T|3D=|3F?|0D |0D |0A |43C|4FO|4DM AT+CPROT=?...COM
4DM|41A|4EN|44D|20 |4EN|4FO|54T|20 |53S|55U|50P|50P|4FO|52R|54T MAND NOT SUPPORT
0D |0A                                                          ..              
Checking for SYNCML/OBEX support
SENDING frame type 0x00/length 0x0C/12
41A|54T|2B+|53S|59Y|4EN|43C|4DM|4CL|3D=|3F?|0D                  AT+SYNCML=?.    
1 "AT+SYNCML=?"
2 "COMMAND NOT SUPPORT"
Checking line: COMMAND NOT SUPPORT
AT reply state: 3
RECEIVED frame type 0x00/length 0x23/35
41A|54T|2B+|53S|59Y|4EN|43C|4DM|4CL|3D=|3F?|0D |0D |0A |43C|4FO AT+SYNCML=?...CO
4DM|4DM|41A|4EN|44D|20 |4EN|4FO|54T|20 |53S|55U|50P|50P|4FO|52R MMAND NOT SUPPOR
54T|0D |0A                                                      T..             
SENDING frame type 0x00/length 0x0D/13
41A|54T|24$|54T|53S|53S|50P|43C|53S|57W|3D=|3F?|0D              AT$TSSPCSW=?.   
1 "AT$TSSPCSW=?"
2 "COMMAND NOT SUPPORT"
Checking line: COMMAND NOT SUPPORT
AT reply state: 3
RECEIVED frame type 0x00/length 0x24/36
41A|54T|24$|54T|53S|53S|50P|43C|53S|57W|3D=|3F?|0D |0D |0A |43C AT$TSSPCSW=?...C
4FO|4DM|4DM|41A|4EN|44D|20 |4EN|4FO|54T|20 |53S|55U|50P|50P|4FO OMMAND NOT SUPPO
52R|54T|0D |0A                                                  RT..            
Setting date & time
SENDING frame type 0x00/length 0x21/33
41A|54T|2B+|43C|43C|4CL|4BK|3D=|22"|322|300|311|366|2F/|300|366 AT+CCLK="2016/06
2F/|300|355|2C,|311|366|3A:|344|399|3A:|333|388|2B+|300|300|22" /05,16:49:38+00"
0D                                                              .               
1 "AT+CCLK="2016/06/05,16:49:38+00""
2 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x27/39
41A|54T|2B+|43C|43C|4CL|4BK|3D=|22"|322|300|311|366|2F/|300|366 AT+CCLK="2016/06
2F/|300|355|2C,|311|366|3A:|344|399|3A:|333|388|2B+|300|300|22" /05,16:49:38+00"
0D |0D |0A |4FO|4BK|0D |0A                                      ...OK..         
Getting firmware versions
SENDING frame type 0x00/length 0x08/8
41A|54T|2B+|43C|47G|4DM|52R|0D                                  AT+CGMR.        
1 "AT+CGMR"
2 "11.314.13.51.156"
3 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x22/34
41A|54T|2B+|43C|47G|4DM|52R|0D |0D |0A |311|311|2E.|333|311|344 AT+CGMR...11.314
2E.|311|333|2E.|355|311|2E.|311|355|366|0D |0A |0D |0A |4FO|4BK .13.51.156....OK
0D |0A                                                          ..              
Received firmware version: "11.314.13.51.156"
Number version is "11,314135"
[Firmware version - "11.314.13.51.156"]
SENDING frame type 0x00/length 0x0A/10
41A|54T|5E^|43C|55U|52R|43C|3D=|300|0D                          AT^CURC=0.      
1 "AT^CURC=0"
2 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x10/16
41A|54T|5E^|43C|55U|52R|43C|3D=|300|0D |0D |0A |4FO|4BK|0D |0A  AT^CURC=0...OK..
SENDING frame type 0x00/length 0x0A/10
41A|54T|2B+|43C|46F|55U|4EN|3D=|311|0D                          AT+CFUN=1.      
1 "AT+CFUN=1"
2 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x10/16
41A|54T|2B+|43C|46F|55U|4EN|3D=|311|0D |0D |0A |4FO|4BK|0D |0A  AT+CFUN=1...OK..
SENDING frame type 0x00/length 0x0D/13
41A|54T|5E^|50P|4FO|52R|54T|53S|45E|4CL|3D=|311|0D              AT^PORTSEL=1.   
1 "AT^PORTSEL=1"
2 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x13/19
41A|54T|5E^|50P|4FO|52R|54T|53S|45E|4CL|3D=|311|0D |0D |0A |4FO AT^PORTSEL=1...O
4BK|0D |0A                                                      K..             
[Connected]
Entering GSM_GetNetworkInfo
Enable full network info
SENDING frame type 0x00/length 0x0A/10
41A|54T|2B+|43C|52R|45E|47G|3D=|322|0D                          AT+CREG=2.      
1 "AT+CREG=2"
2 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x10/16
41A|54T|2B+|43C|52R|45E|47G|3D=|322|0D |0D |0A |4FO|4BK|0D |0A  AT+CREG=2...OK..
Enable full packet network info
SENDING frame type 0x00/length 0x0B/11
41A|54T|2B+|43C|47G|52R|45E|47G|3D=|322|0D                      AT+CGREG=2.     
1 "AT+CGREG=2"
2 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x11/17
41A|54T|2B+|43C|47G|52R|45E|47G|3D=|322|0D |0D |0A |4FO|4BK|0D  AT+CGREG=2...OK.
0A                                                              .               
Getting GPRS state
SENDING frame type 0x00/length 0x0A/10
41A|54T|2B+|43C|47G|41A|54T|54T|3F?|0D                          AT+CGATT?.      
1 "AT+CGATT?"
2 "+CGATT: 0"
3 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x1D/29
41A|54T|2B+|43C|47G|41A|54T|54T|3F?|0D |0D |0A |2B+|43C|47G|41A AT+CGATT?...+CGA
54T|54T|3A:|20 |300|0D |0A |0D |0A |4FO|4BK|0D |0A              TT: 0....OK..   
GPRS state received
Parsing +CGATT: 0 with +CGATT: @i
Parsed int 0
Getting network LAC and CID and state
SENDING frame type 0x00/length 0x09/9
41A|54T|2B+|43C|52R|45E|47G|3F?|0D                              AT+CREG?.       
1 "AT+CREG?"
2 "+CREG: 2,2,45B3,41C"
3 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x26/38
41A|54T|2B+|43C|52R|45E|47G|3F?|0D |0D |0A |2B+|43C|52R|45E|47G AT+CREG?...+CREG
3A:|20 |322|2C,|322|2C,|344|355|42B|333|2C,|344|311|43C|0D |0A  : 2,2,45B3,41C..
0D |0A |4FO|4BK|0D |0A                                          ..OK..          
Network LAC & CID & state received
Parsing +CREG: 2,2,45B3,41C with +CREG: @i, @i, @r, @r, @i
Parsed int 2
Parsed int 2
Grabbed string from reply: "45B3" (parsed 4 bytes)
Parsed raw string "45B3"
Grabbed string from reply: "41C" (parsed 3 bytes)
Parsed raw string "41C"
Parsing +CREG: 2,2,45B3,41C with +CREG: @i, @i, @r, @r
Parsed int 2
Parsed int 2
Grabbed string from reply: "45B3" (parsed 4 bytes)
Parsed raw string "45B3"
Grabbed string from reply: "41C" (parsed 3 bytes)
Parsed raw string "41C"
Not registered into any network. Searching for network
Getting packet network LAC and CID and state
SENDING frame type 0x00/length 0x0A/10
41A|54T|2B+|43C|47G|52R|45E|47G|3F?|0D                          AT+CGREG?.      
1 "AT+CGREG?"
2 "+CGREG: 2,2,45B3,41C"
3 "OK"
Checking line: OK
AT reply state: 1
RECEIVED frame type 0x00/length 0x28/40
41A|54T|2B+|43C|47G|52R|45E|47G|3F?|0D |0D |0A |2B+|43C|47G|52R AT+CGREG?...+CGR
45E|47G|3A:|20 |322|2C,|322|2C,|344|355|42B|333|2C,|344|311|43C EG: 2,2,45B3,41C
0D |0A |0D |0A |4FO|4BK|0D |0A                                  ....OK..        
Network LAC & CID & state received
Parsing +CGREG: 2,2,45B3,41C with +CGREG: @i, @i, @r, @r, @i, @r
Parsed int 2
Parsed int 2
Grabbed string from reply: "45B3" (parsed 4 bytes)
Parsed raw string "45B3"
Grabbed string from reply: "41C" (parsed 3 bytes)
Parsed raw string "41C"
Parsing +CGREG: 2,2,45B3,41C with +CGREG: @i, @i, @r, @r
Parsed int 2
Parsed int 2
Grabbed string from reply: "45B3" (parsed 4 bytes)
Parsed raw string "45B3"
Grabbed string from reply: "41C" (parsed 3 bytes)
Parsed raw string "41C"
Not registered into any network. Searching for network
Leaving GSM_GetNetworkInfo
[Terminating]
[Closing]
```

**Comment by PaleoCrafter** [comment]
I was able to resolve the issue myself by configuring the modem via the vendor's software. I had to switch the device to not only search for 3G networks (or maybe that's just what the software told it to look for) and I had to search and register the network manually. Thanks God the configuration was saved on the stick, otherwise I would not have been able to complete my project.

If this configuration can somehow be performed by Gammu, I'd appreciate to learn about it. If it is in fact not capable of doing it, consider this a feature request 😊
